### PR TITLE
More release improvements from v0.15.0

### PR DESCRIPTION
**Pull request by jedevc**
Splitting out some of the additional tasks from https://github.com/dagger/dagger/pull/9167

### CODE REVIEW

**Comment by gerhard** [comment]
Are [these lint failures](https://dagger.cloud/dagger/traces/79ce0f4d7134f5235f8945633b45e848) addressed in a separate PR @jedevc ?

image

**Comment by jedevc** [comment]
Nope, they're fixed now :laughing:

File: modules/daggerverse/main.go
Comment (gerhard) [inline modules/daggerverse/main.go:1]: LGTM!

Just to double-check:
- the author of this PR will be `dagger-ci` (including assignee)
- the reviewer will be the release engineer (you in the case of `v0.15.0`)

**Review by gerhard — APPROVED** [review]
Left a few nits, no reason to block the merge over them.

Look forward to confirming that this works as expected **after** we merge it.

File: modules/daggerverse/main.go
Comment (jedevc) [reply modules/daggerverse/main.go:1]: Yup, that's what should happen now! Only the assignee has been changed, so everything will work exactly as before, except for that bit.

File: .dagger/cli.go
Comment (jedevc) [reply .dagger/cli.go:213]: Agreed this is cruft, but it's cruft we'll probably use again - IMO, having this inline for when we inevitably need it next (probably the next go release) is better than having to come back and attempt to reconstruct the logic. I want to try and keep as much tribal knowledge out of the release + maintenance process - the more tribal knowledge we have, the more it rests on just a couple of people.

Actually, the idea is to remove this logic entirely. As part of the release improvements, I want to have goreleaser ingest the completely built binaries, instead of attempting to build them itself - it's frustrating to have the discrepancy we already have (and it's required if we want to persue the one-binary goal). With that, we can entirely remove this snippet.

**Pull request merged**
